src/utils/responseHandlers.ts

```diff
@@ -3,51 +3,58 @@ import { ApiResponse } from './apiResponse';
 
 class ResponseHandler {
   toApiResponse(promise: Promise<any>) {
-    return promise.then((response) => {
-      if (!response) {
-        return Promise.reject(new Error('Request failed.'));
-      }
-      if (response.status === 200 && response.data.Success) {
-        return response.data;
-      }
-      // response.statusText is never null,
-      // rather an empty string is returned on eg. 403 => fallback to status code
-      const message = response.statusText || response.status.toString();
-      return Promise.reject(new Error(message));
-    });
+    return promise
+      .then((response: AxiosResponse) => {
+        if (response.data.Success) {
+          return response.data;
+        }
+        return Promise.reject(response);
+      })
+      .catch((error: any) => {
+        if (error.response) {
+          return Promise.reject(error.resonse);
+        }
+        return Promise.reject(new Error(error.message));
+      });
   }
 
   /**
    * @return A promise that resolves to T or rejects if no results
    */
   required<T>(promise: Promise<AxiosResponse>): Promise<T> {
-    return this.toApiResponse(promise).then((response: ApiResponse<T[]>) => {
-      if (response.NumResults > 0) {
-        return response.Results[0];
-      }
-      return Promise.reject(new Error(response.ErrorMessage!));
-    });
+    return this.toApiResponse(promise)
+      .then((response: ApiResponse<T[]>) => {
+        if (response.NumResults > 0) {
+          return response.Results[0];
+        }
+        return Promise.reject(new Error(response.ErrorMessage!));
+      })
+      .catch((error) => error);
   }
 
   requiredList<T>(promise: Promise<AxiosResponse>): Promise<T[]> {
```

Comment:
This is now conflicting with required method, so i would have required, requiredList (has to return entity or list of entities) and than 2 for not requiring (maybe just passed as a param instead of own function). What do you think?

Reply:
Yes, we need to clean up the  required method also the optional can be removed but I didn't want to do this in this PR